### ISSUE TITLE
Scanning for buttons

### ISSUE THREAD

**Issue by klaas** [50ButtonsEach/flic2lib-ios#8]
Hej there,

I tried to follow the tutorial at https://github.com/50ButtonsEach/flic2lib-ios/wiki/Tutorial
(https://partners.flic.io/partners/developers/ios-tutorial seems outdated, but has more stuff in it like configuring URL schemes.), but have no success in using the flic buttons.

Everything compiles fine after configuring a framework search path (`FRAMEWORK_SEARCH_PATHS`) in my build settings.

I added a simple button to my app to trigger the `scanForButtonsWithStateChangeHandler` method. But the only thing I get is one of these two errors in the completion closure:

`The operation couldn’t be completed. (com.shortcutlabs.flic2lib.buttonscanner error 2.)`

`The operation couldn’t be completed. (com.shortcutlabs.flic2lib.buttonscanner error 7.)`

I'm running everything on the device and not in the simulator. The Flic app is running. I added some buttons to my hub and some to the iPhone and some are still unpaired. Triggering lights and playing sounds through the hub is working very nicely.

Any help for setting up a sample iOS project would be appreciated.

**Comment by AntonMeier** [comment]
We will release an example project soon. But until then I can only say that error codes 2 and 7 are NoPublicButtonDiscovered and UserCanceled, respectively. The first one means that no Flic2 button was discovered during that scan and the second one means that the iOS bluetooth pairing dialog was not accepted.

Regarding documentation, the one on our website is for the old Flic1 buttons, while the one on github is for the new Flic2. 

What kind of button are you using? Also, the button that you wish to pair, are you sure that it is not currently connected to a hub?

**Comment by klaas** [comment]
@AntonMeier Thanks for your fast reply! I read the Flic1 tutorial first and thought that the Flic2 would work similar. I was under the impression that after calling `scanForButtons` something would have to happen. Sorry, that was my misunderstanding. I'm new to the Flic buttons.

Just got my Flic2 buttons and a LR hub some days ago.

After some more poking I can see now, that I can add buttons to my app by long-pressing them during the `scanForButtons` call. This works with buttons that are still unpaired or that are already paired with my iPhone. This does not work for buttons that are paired to the hub. 😃

Thanks or providing the error codes. I now found them myself in `FLICEnums.h`. Having abandoned Objctive-C some years ago I forget to check that there might be a list of error codes. Pretty used to getting a description from `NSError.localizedDescription` nowadays :-) (no criticism, just feedback of my experience using the API for the first time.)

Because I'm using Swift I found a small issue, regarding the nullability of the completion block `button` argument. My tests show that the button might be nil, but the headers do not mark them as `_NULLABLE`.

Thanks again for the fast reply and pointing me into the right direction! I really like how fast the buttons work! And the ability to fine tune the configuration with the `FLICButtonTriggerMode` is awesome!
